server/chat-plugins/datasearch.ts

```diff
@@ -10,15 +10,59 @@
 
 'use strict';
 
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {Chat} from '../chat';
+import * as CL from '../config-loader';
+import {Dex} from '../../sim/dex';
+import {TeamValidator} from '../../sim/team-validator';
+
+interface DexOrGroup {
+	abilities: {[k: string]: boolean};
+	tiers: {[k: string]: boolean};
+	doublesTiers: {[k: string]: boolean};
+	colors: {[k: string]: boolean};
+	'egg groups': {[k: string]: boolean};
+	formes: {[k: string]: boolean};
+	gens: {[k: string]: boolean};
+	moves: {[k: string]: boolean};
+	types: {[k: string]: boolean};
+	resists: {[k: string]: boolean};
+	weak: {[k: string]: boolean};
+	stats: {[k: string]: {[k in Direction]: number}};
+	skip: boolean;
+}
+
+interface MoveOrGroup {
+	types: {[k: string]: boolean};
+	categories: {[k: string]: boolean};
+	contestTypes: {[k: string]: boolean};
+	flags: {[k: string]: boolean};
+	gens: {[k: string]: boolean};
+	recovery: {[k: string]: boolean};
+	mon: {[k: string]: boolean};
+	property: {[k: string]: {[k in Direction]: number}};
+	boost: {[k: string]: boolean};
+	lower: {[k: string]: boolean};
+	zboost: {[k: string]: boolean};
+	status: {[k: string]: boolean};
+	volatileStatus: {[k: string]: boolean};
+	recoil: boolean;
+	skip: boolean;
+}
+
+type Direction = 'less' | 'greater' | 'equal';
+type StatAliases = 'attack' | 'defense' | 'specialattack' | 'spc' | 'special' | 'spatk' | 'specialdefense' | 'spdef' | 'speed' | 'wt' | 'ht' | 'generation';
```

Comment:
Is this necessary? As far as I can tell, this provides zero type safety, since you just cast straight past this.

Reply:
bump

Reply:
nope, removed